### PR TITLE
DDF-3380 Made UI Branding Name customizable

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
This PR makes the UI Name configurable.  It also cleans up the ConfigurationApplication.java class.
#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@adimka @Variadicism 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@rzwiefel
#### How should this be tested? (List steps with links to updated documentation)
Build / Install / Change UI Name / Observe Changes
#### What are the relevant tickets?
[DDF-3380](https://codice.atlassian.net/browse/DDF-3380)
#### Screenshots (if appropriate)
screen shot 2017-10-13 at 9 11 43 am
screen shot 2017-10-13 at 9 11 53 am
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java [hunk 1]
```diff
@@ -342,6 +344,7 @@ public void destroy() {
     config.put("isExperimental", isExperimental);
     config.put("autoMergeTime", autoMergeTime);
     config.put("mapHome", mapHome);
+    config.put("product", uiName);
```
Comment (Variadicism) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java]: :question: This property seems to be called `uiName` everywhere else; why is it called "`product`" in the configuration?

Comment (bdeining) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java]: The front end uses the name `product`; this was easier than refactoring every place `product` is used.

**Comment by mojogitoverhere** [comment]
build now

**Comment by mojogitoverhere** [comment]
Hero build was successful! 🎉 
Steps:
1. Quick build
2. Manually verify "Intrigue" is updated in UI

Results:
- Empty strings work (does not default to Intrigue, everything just says "DDF")
- Very large strings work (title bar becomes a side scrolling box but the layout remains intact)
- Regular strings work

**Pull request merged**